gno.land/pkg/gnoclient/client_txs.go

```diff
@@ -147,17 +180,21 @@ func (c *Client) Run(cfg BaseTxCfg, msgs ...MsgRun) (*ctypes.ResultBroadcastTxCo
 			return nil, err
 		}
 
-		caller, err := c.Signer.Info()
-		if err != nil {
-			return nil, err
+		callerAddress := cfg.CallerAddress
+		if callerAddress.IsZero() {
+			caller, err := c.Signer.Info()
+			if err != nil {
+				return nil, err
+			}
+			callerAddress = caller.GetAddress()
 		}
 
 		msg.Package.Name = "main"
 		msg.Package.Path = ""
 
 		// Unwrap syntax sugar to vm.MsgCall slice
 		vmMsgs = append(vmMsgs, std.Msg(vm.MsgRun{
-			Caller:  caller.GetAddress(),
+			Caller:  callerAddress,
```

Comment:
Same comment as before, for `MsgSend`

Reply:
Done. As with `MsgSend`, after merging PR #2715, these functions don't need to set the `Caller`.